### DISCUSSION TITLE
How to obtain the length of the current task queue from the runtime

### DISCUSSION THREAD

**Discussion by konghaiya** [tokio-rs/tokio#6485] [Q&A]
How to obtain the length of the current task queue from the runtime，Or how to know how long a task has been queued in the queue

**Comment by konghaiya** [comment]
We are conducting performance pressure tests and suspect that a significant amount of time is spent on task queuing, but the time spent on this part cannot be counted

**Comment by Darksonn** [comment]
Perhaps [`RuntimeMetrics`][1] fits your user-case? There is [`injection_queue_depth`](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.injection_queue_depth) and [`worker_local_queue_depth`](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html#method.worker_local_queue_depth).

[1]: https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html

**Reply by konghaiya** [reply]
let metrics = Handle::current().metrics();   What version does this metric start with? It seems that the 1.27 version I am using does not have this method

**Reply by Darksonn** [reply]
It is present on Tokio 1.27, but you need to [enable unstable Tokio features](https://docs.rs/tokio/latest/tokio/index.html#unstable-features) to use it.

**Reply by konghaiya** [reply]
Okay, thank you very much
